### PR TITLE
Fundament changes to the Database

### PR DESCRIPTION
**Pull request by EricKarpovits**
- Changed the Database so that it takes in a unique string identifier (Username) and an ArrayList of Objects - Such as the User and Fridge as specified. 
Let me know if there is something I should change :).

### CODE REVIEW

**Review by Shaffaan — APPROVED** [review]
Looks good eric, this should now work with the use cases!

File: src/main/java/main.java [hunk 1]
```diff
@@ -1,5 +1,30 @@
+import database.Database;
+import database.DatabaseGateway;
+import fridge.Fridge;
+import fridge.Ingredient;
+
+import java.util.ArrayList;
+
 public class main {
     public static void main(String[] args) {
+        Fridge fridgeOne = new Fridge();
```
Comment (fishyvishy) [inline src/main/java/main.java:10]: confusing variable name

**Review by fishyvishy — APPROVED** [review]
Seems solid. Will patch it with startPage UI and register use case.

File: src/main/java/main.java (hunk 1 above)
Comment (EricKarpovits) [reply src/main/java/main.java:10]: Fixed

**Review by rickychen175 — APPROVED** [review]
The database looks good.

**Pull request merged**